lib/rubocop/ast/node.rb

```diff
@@ -510,8 +510,14 @@ def guard_clause?
 
       # @!method class_constructor?(node = self)
       def_node_matcher :class_constructor?, <<~PATTERN
-        {       (send #global_const?({:Class :Module :Struct}) :new ...)
-         (block (send #global_const?({:Class :Module :Struct}) :new ...) ...)}
+        {
+          (send #global_const?({:Class :Module :Struct}) :new ...)
+          (send #global_const?(:Data) :define ...)
+          ({block numblock} {
```

Comment:
This is good, but makes me wonder if we shouldn't have a pseudo type to include both of these... Maybe `any_block` or `a_block` instead of `{block numblock}`